peer.go

```diff
@@ -318,7 +318,7 @@ func newPeerScore(p *Peer, score uint64) *peerScore {
 type Peer struct {
 	sync.RWMutex
 
-	channel             Connectable
```

Comment:
Rather than changing this reference to use `*Channel` and reaching into the `*Channel` from the peer, I think we should encapsulate the logic within the peer itself.

The `newPeer` constructor should take a `onPeerConnChange func(*Peer)`, and same for the root peer list. that way, it's not coupled to the channel, but uses a more DI style approach.